core/src/commonMain/kotlin/com/ustadmobile/core/account/UstadAccountManager.kt

```diff
@@ -505,6 +526,56 @@ class UstadAccountManager(
         }
     }
 
+    suspend fun createLocalAccount(): UserSessionWithPersonAndEndpoint {
+        // Create a random IP and endpoint
+        val randomIp = "169.${(0..255).random()}.${(0..255).random()}.${(0..255).random()}"
+        val randomPort = (1000..9999).random()
+        val fakeEndpoint = Endpoint("http://$randomIp:$randomPort/")
+
+        val db: UmAppDatabase by di.on(fakeEndpoint).instance(tag = DoorTag.TAG_DB)
+
+        return db.withDoorTransactionAsync {
+            val newSite = Site().apply {
+                siteName = "Local Site"
+                authSalt = "local_${randomString(10)}"
+            }
+            db.siteDao.insert(newSite)
+
+            val newPerson = Person().apply {
+                personUid = db.doorPrimaryKeyManager.nextId(Person.TABLE_ID)
+                username = "local_user_${randomString(5)}"
+                firstNames = "Local"
+                lastName = "User"
+                personType = Person.TYPE_GUEST
+            }
+            db.personDao.insert(newPerson)
+
+            val newSession = UserSession().apply {
+                usUid = db.doorPrimaryKeyManager.nextId(UserSession.TABLE_ID)
+                usClientNodeId = db.doorWrapperNodeId
+                usStartTime = systemTimeInMillis()
+                usSessionType = UserSession.TYPE_TEMP_LOCAL or UserSession.TYPE_GUEST
+                usStatus = UserSession.STATUS_ACTIVE
+                usPersonUid = newPerson.personUid
+            }
+            db.userSessionDao.insertSession(newSession)
+
+            UserSessionWithPersonAndEndpoint(newSession, newPerson, fakeEndpoint)
+        }.also { newSession ->
+            addActiveEndpoint(fakeEndpoint)
+            currentUserSession = newSession
+        }
+    }
+
+    private fun randomString(length: Int): String {
```

Comment:
We already have a randomString function - see RandomString.kt . Please use the existing function.